src/main/java/Assignment1.java

```diff
@@ -0,0 +1,175 @@
+            return false;
+        }
+        if (price.isEmpty()) {
+            System.out.println("Invalid price");
+            return false;
+        }
+        if (quantity.isEmpty()) {
+            System.out.println("Invalid quantity");
+            return false;
+        }
+        if (type.isEmpty()) {
+            System.out.println("Invalid type");
+            return false;
+        }
+        // name should only have alphabets.
+        if (!name.matches("^[ A-Za-z]+$")) {
+            System.out.println("Invalid name");
+            return false;
+        }
+        // quantity should only have numbers.
+        if (!quantity.matches("[0-9]*") || Integer.parseInt(quantity) <= 0) {
+            System.out.println("Invalid quantity");
+            return false;
+        }
+        // price should only have numbers and can have decimal (.) once.
+        if (!price.matches("[0-9.]*") || (price.length() - price.replace(".", "").length() > 1)) {
+            System.out.println("Invalid price");
+            return false;
+        }
+        // type check.
+        if (!type.equals("raw") && !type.equals("manufactured") && !type.equals("imported")) {
+            System.out.println("Invalid type");
+            return false;
+        }
+        return true;
+    }
+
+    // method to extract data from input to member variables
+    void extractData(String input_string) {
+        String name = extractWord(input_string, input_string.indexOf("-name") + 5).trim();
+        String quantity = extractWord(input_string, input_string.indexOf("-quantity") + 9).trim();
+        String price = extractWord(input_string, input_string.indexOf("-price") + 6).trim();
+        String type = extractWord(input_string, input_string.indexOf("-type") + 5).trim();
+        _name = name;
+        _quantity = Integer.parseInt(quantity);
+        _price = Double.parseDouble(price);
+        _type = type;
+    }
+
+    // method to calculate raw tax.
+    double rawTax(double total) {
+        if (total < 0) throw new RuntimeException("Total cant be -ve");
+        double tax = 0;
+        tax += total * 12.50 / 100;
+        return tax;
+    }
+
+    // method to calculate manufactured tax.
+    double manufacturedTax(double total) {
+        if (total < 0) throw new RuntimeException("Total cant be -ve");
+        double tax = 0;
+        tax += total * 12.5 / 100;
+        tax += (total + tax) * 2.0 / 100;
+        return tax;
+    }
+
+    // method to calculate imported tax.
+    double importedTax(double total) {
+        if (total < 0) throw new RuntimeException("Total cant be -ve");
+        double tax = total * 0.1;
+        if (total <= 100)
+            tax += 5;
+        else if (total <= 200)
+            tax += 10;
+        else
+            tax += (total + tax) * 0.05;
+        return tax;
+    }
+
+    // method to calculate total cost.
+    double calculateCost() {
+        double totalCost = _price * _quantity;
```

Comment:
no need to start with _ for private member variables.

Reply:
done.